examples/examples/z_formats.rs

```diff
@@ -0,0 +1,16 @@
+use zenoh::{keformat, prelude::keyexpr};
+
+keformat!("user_id/${user_id:*}/file/${file:**}" mod format);
+
+fn main() {
+    let mut formatter: format::Formatter = format::formatter();
+    let file = "hi/there";
+    let ke = keformat!(formatter, user_id = 42, file)
+        .unwrap()
+        .build()
+        .unwrap();
+    println!("{formatter:?} => {ke}");
+    let parsed = format::parse(&ke).unwrap();
```

Comment:
Why there is the need to parse the `ke` if it has just been built at line 8?

Reply:
Can I use it to retrieve a `KeyExpr` e.g. from a `Sample` in a subscriber callback, parse it, and access the field? 
I believe so, but I think it would be better to be more explicit in the example.

Reply:
I changed the example to parse an arbitrary KE with a different format.

I avoided making subscribers/queryables to avoid distracting from the formatting aspects.